src/utils/proxy.ts

```diff
@@ -365,7 +365,7 @@ const serveRedirect = async function ({
     (await isEndpointExists(decodeURIComponent(reqUrl.pathname), options.target))
   if (staticFile || endpointExists) {
     const pathname = staticFile || reqUrl.pathname
-    req.url = encodeURI(pathname) + reqUrl.search
+    req.url = encodeURI(decodeURI(pathname)) + reqUrl.search
```

Comment:
why `decodeURIComponent` on line 365 and `decodeURI` here? can we use the same one and share the result?

Reply:
Let's iterate on this, I need this released faster, and this micro optimization not worth the additional testing effort